solr/core/src/test/org/apache/solr/cloud/SyncSliceTest.java

```diff
@@ -217,11 +218,13 @@ private void waitTillAllNodesActive() throws Exception {
       ZkStateReader zkStateReader = cloudClient.getZkStateReader();
       ClusterState clusterState = zkStateReader.getClusterState();
       DocCollection collection1 = clusterState.getCollection("collection1");
+      ShardStateProvider ssp = zkStateReader.getShardStateProvider(collection1.getName());
+
       Slice slice = collection1.getSlice("shard1");
       Collection<Replica> replicas = slice.getReplicas();
       boolean allActive = true;
       for (Replica replica : replicas) {
-        if (!clusterState.liveNodesContain(replica.getNodeName()) || replica.getState() != Replica.State.ACTIVE) {
+        if (!clusterState.liveNodesContain(replica.getNodeName()) || ssp.getState(replica)!= Replica.State.ACTIVE) {
```

Comment:
Why not use a `!ssp.isActive(replica)`?

I see inconsistent usage of this logic throughout.

Reply:
OK